src/core/data-type.js

```diff
@@ -0,0 +1,41 @@
+// @ts-check
+/**
+ * Module core/data-type
+ * Propagates data type of a <var> in all occurrences within a section.
+ * Also adds the CSS for the data type tooltip.
+ * Set `conf.highlightVars = true` to enable.
+ */
+import tooltipStyle from "text!../../assets/datatype.css";
+
+export const name = "core/data-type";
+
+export function run(conf) {
+  if (!conf.highlightVars) {
+    return;
+  }
+
+  const style = document.createElement("style");
+  style.textContent = tooltipStyle;
+  document.head.appendChild(style);
+
+  document.querySelectorAll("section").forEach(section => {
```

Comment:
Alternatively, query “section var”, then you can check if you are still in the same section by using something like varElement.closest(“section”) === lastSeenSection. 

You could also keep a map per section, to quickly look up a var’s type by name.

Reply:
I don't know, querying `var` separately for each section seems more intuitive to me than querying `section var`. Any benefits of querying “section var” I am missing? 

The map idea is great, will do.

Reply:
> Any benefits of querying “section var” I am missing?

if flattens the algorithm, but the trade off is then needing to run `.closest()`

So, something like:
 
```JS
let section = null;
const varMap = new Map();
for (const varElem of document.querySelectorAll("section var")) {
  const currentSection = varElem.closest("section");
  if (lastSeenSection !== currentSection) {
     section = currentSection;
     varMap.clear();
  }
  if (varElement.data.type) {
     varMap.add(varElement.textContent.trim(), varElement.data.type);
     continue;
  }
  const type = varMap.get(varElement.textContent.trim());
  if (type) var.dataset.type = type;
}
```

Reply:
fixed some typos above.